### ISSUE TITLE
Clicking on canvas during Inspect with selected node highlights the wrong thing

### ISSUE THREAD

**Issue by cdsmith** [google/codeworld#1425]
To reproduce:

1, Run a program with lots of large translated shapes over most of the screen.
2. Click Inspect.
3. Highlight a leaf node of some kind underneath a `translated` function, so that the canvas focuses on just that node and draws it at its untranslated location.
4. Click on the canvas.

* Expected behavior: The shape that you clicked on, if any, in the focused picture becomes the new focus.
* Actual behavior: The shape at that coordinate in the **original** picture becomes the new focus.

Basically, it looks like we're doing hit testing on the wrong picture!

**Comment by cdsmith** [comment]
I looked into this briefly.  It's not trivial as easy as just swapping one picture for another, because node ids in a filtered picture are different from those in a non-filtered picture.  So what needs to be done is to find the root node id at the base of the filtered picture, compute the node id in the filtered picture, and then adjust to produce a global node id.  To do this, one needs not just the picture, but also the node id.

There are three layers here, and only the first and third is aware of the current node id.

* Layer 1: The inspect UI in the parent frame, implemented in `codeExplorer.js`
* Layer 2: The debug support code in the running program environment, implemented in `run.js` and low-level parts of `CodeWorld.Driver`.
* Layer 3: Debug state management in the main program loop, implemented in `CodeWorld.Driver`.

Layer 1 is the source of truth for the currently selected node.  Layer 3 is aware of the currently selected node, but may not be up-to-date, since it is notified asynchronously.  Layer 2, where we need the current node id, is not aware of it at all.  I think the solution here is to make Layer 2 aware of the currently selected node (again, via asynchronous notification), so it can pass it in to the `getNode` endpoint.
